src/go_database_dsn.rs

```diff
@@ -103,6 +103,7 @@ static DSN_REGEX: Lazy<Regex> = Lazy::new(|| {
 pub(crate) struct GoDatabaseDsn {
     username: Option<String>,
     password: Option<String>,
+    protocol: String,
```

Comment:
If I read this PR correctly, I think you're just supporting `tcp` and `unix` as protocols. It may be worth making this an enum. This would make the line `if self.protocol == "unix"` be more like `match self.protocol { Protocol::Unix => {...}, Protocol::TCP => { ... } }` which is cleaner. Looking at the [list of mysql protocols](https://dev.mysql.com/doc/refman/8.0/en/transport-protocols.html#transport-protocol-selection), unix and tcp are the ones you'll care about so it's probably not worth it.

Reply:
I considered that, but decided not to for that same reason.